In [1]:
from sklearn.svm import SVR
from sklearn.preprocessing import StandardScaler
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.metrics import plot_roc_curve
from sklearn.metrics import roc_auc_score, roc_curve, auc
from sklearn.metrics import classification_report, confusion_matrix, accuracy_score
from sklearn.model_selection import KFold, GridSearchCV
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error

# 데이터 불러오기

## 효준이 감성 점수 포함된 데이터 불러오기

In [32]:
df = pd.read_csv('분석용데이터_utf_spi수정완료.csv')

In [33]:
df2 = pd.read_csv('df_verb.csv')

In [34]:
df2

Unnamed: 0      M2  comp basemoney rate  base rate  consumer price index  \
0             0  1676.4                 4.52       2.75                   3.4   
1             1  1674.4                 4.72       2.75                   3.9   
2             2  1677.5                 4.54       3.00                   4.1   
3             3  1684.8                 4.54       3.00                   3.8   
4             4  1690.5                 4.44       3.00                   3.9   
..          ...     ...                  ...        ...                   ...   
115         115  3100.4                 2.19       0.50                   0.7   
116         116  3115.2                 2.24       0.50                   1.0   
117         117  3152.8                 2.24       0.50                   0.1   
118         118  3183.5                 2.25       0.50                   0.6   
119         119  3183.5                 2.21       0.50                   0.5   

     kospi close  ex-rate close  next_spi  sieon_sent_score  
0        2069.73        1119.25      82.5     -1.635618e+08  
1        1939.30        1124.65      83.4     -1.777621e+08  
2        2106.70        1095.50      83.9     -1.980306e+08  
3        2192.36        1068.00      84.1     -1.898849e+08  
4        2142.47        1078.00      84.4     -2.402702e+08  
..           ...            ...       ...               ...  
115      2326.17        1187.94     114.0      3.239718e+07  
116      2327.89        1164.65     115.6      4.511543e+07  
117      2267.15        1136.49     118.3      4.169356e+07  
118      2591.34        1109.32     121.4      4.231788e+07  
119      2873.47        1084.47     124.9      5.645645e+07  

[120 rows x 9 columns]

In [35]:
b = df2[['sieon_sent_score']]

In [48]:
b.index = df.index

In [49]:
b

sieon_sent_score
Date                     
201101월     -1.635618e+08
201102월     -1.777621e+08
201103월     -1.980306e+08
201104월     -1.898849e+08
201105월     -2.402702e+08
...                   ...
202008월      3.239718e+07
202009월      4.511543e+07
202010월      4.169356e+07
202011월      4.231788e+07
202012월      5.645645e+07

[120 rows x 1 columns]

In [36]:
#Date를 index로 넣어줌
df.index = df['Date']
df.drop(['Date'], axis = 1, inplace = True)

In [50]:
df

sent_score      M2  comp basemoney rate  base rate  \
Date                                                            
201101월 -6.268386e+07  1676.4                 4.52       2.75   
201102월 -6.255832e+07  1674.4                 4.72       2.75   
201103월 -6.957441e+07  1677.5                 4.54       3.00   
201104월 -7.305733e+07  1684.8                 4.54       3.00   
201105월 -9.356633e+07  1690.5                 4.44       3.00   
...               ...     ...                  ...        ...   
202008월 -5.824726e+07  3100.4                 2.19       0.50   
202009월 -6.441351e+07  3115.2                 2.24       0.50   
202010월 -3.751069e+07  3152.8                 2.24       0.50   
202011월 -3.147741e+07  3183.5                 2.25       0.50   
202012월 -3.986893e+07  3183.5                 2.21       0.50   

         consumer price index  kospi close  ex-rate close       효준감성점수    spi  
Date                                                                           
201101월                   3.4      2069.73        1119.25  1856.746125   81.3  
201102월                   3.9      1939.30        1124.65  2046.702906   82.5  
201103월                   4.1      2106.70        1095.50  1715.795289   83.4  
201104월                   3.8      2192.36        1068.00  1635.262791   83.9  
201105월                   3.9      2142.47        1078.00  1922.518709   84.1  
...                       ...          ...            ...          ...    ...  
202008월                   0.7      2326.17        1187.94  1221.939746  113.1  
202009월                   1.0      2327.89        1164.65  1208.597571  114.0  
202010월                   0.1      2267.15        1136.49  1001.099665  115.6  
202011월                   0.6      2591.34        1109.32   919.546249  118.3  
202012월                   0.5      2873.47        1084.47  1110.592804  121.4  

[120 rows x 9 columns]

## 시언이 감성 점수 불러오기

In [38]:
temp = pd.read_csv('news_with_sent_score.csv')
temp.drop(['Unnamed: 0'], axis = 1, inplace = True)

In [39]:
sent_score = temp['sent_score']

In [40]:
sent_score

0     -6.268386e+07
1     -6.255832e+07
2     -6.957441e+07
3     -7.305733e+07
4     -9.356633e+07
           ...     
115   -5.824726e+07
116   -6.441351e+07
117   -3.751069e+07
118   -3.147741e+07
119   -3.986893e+07
Name: sent_score, Length: 120, dtype: float64

In [41]:
del temp

## 효준이 감성 점수 포함 데이터 프레임에 시언이 감성 점수 포함시키기

In [42]:
sent_score.index = df.index

In [43]:
df = pd.concat([sent_score, df], axis = 1)

In [44]:
df

sent_score      M2  comp basemoney rate  base rate  \
Date                                                            
201101월 -6.268386e+07  1676.4                 4.52       2.75   
201102월 -6.255832e+07  1674.4                 4.72       2.75   
201103월 -6.957441e+07  1677.5                 4.54       3.00   
201104월 -7.305733e+07  1684.8                 4.54       3.00   
201105월 -9.356633e+07  1690.5                 4.44       3.00   
...               ...     ...                  ...        ...   
202008월 -5.824726e+07  3100.4                 2.19       0.50   
202009월 -6.441351e+07  3115.2                 2.24       0.50   
202010월 -3.751069e+07  3152.8                 2.24       0.50   
202011월 -3.147741e+07  3183.5                 2.25       0.50   
202012월 -3.986893e+07  3183.5                 2.21       0.50   

         consumer price index  kospi close  ex-rate close       효준감성점수    spi  
Date                                                                           
201101월                   3.4      2069.73        1119.25  1856.746125   81.3  
201102월                   3.9      1939.30        1124.65  2046.702906   82.5  
201103월                   4.1      2106.70        1095.50  1715.795289   83.4  
201104월                   3.8      2192.36        1068.00  1635.262791   83.9  
201105월                   3.9      2142.47        1078.00  1922.518709   84.1  
...                       ...          ...            ...          ...    ...  
202008월                   0.7      2326.17        1187.94  1221.939746  113.1  
202009월                   1.0      2327.89        1164.65  1208.597571  114.0  
202010월                   0.1      2267.15        1136.49  1001.099665  115.6  
202011월                   0.6      2591.34        1109.32   919.546249  118.3  
202012월                   0.5      2873.47        1084.47  1110.592804  121.4  

[120 rows x 9 columns]

In [52]:
final_df = pd.concat([df,b], axis = 1)

In [54]:
final_df

sent_score      M2  comp basemoney rate  base rate  \
Date                                                            
201101월 -6.268386e+07  1676.4                 4.52       2.75   
201102월 -6.255832e+07  1674.4                 4.72       2.75   
201103월 -6.957441e+07  1677.5                 4.54       3.00   
201104월 -7.305733e+07  1684.8                 4.54       3.00   
201105월 -9.356633e+07  1690.5                 4.44       3.00   
...               ...     ...                  ...        ...   
202008월 -5.824726e+07  3100.4                 2.19       0.50   
202009월 -6.441351e+07  3115.2                 2.24       0.50   
202010월 -3.751069e+07  3152.8                 2.24       0.50   
202011월 -3.147741e+07  3183.5                 2.25       0.50   
202012월 -3.986893e+07  3183.5                 2.21       0.50   

         consumer price index  kospi close  ex-rate close       효준감성점수    spi  \
Date                                                                            
201101월                   3.4      2069.73        1119.25  1856.746125   81.3   
201102월                   3.9      1939.30        1124.65  2046.702906   82.5   
201103월                   4.1      2106.70        1095.50  1715.795289   83.4   
201104월                   3.8      2192.36        1068.00  1635.262791   83.9   
201105월                   3.9      2142.47        1078.00  1922.518709   84.1   
...                       ...          ...            ...          ...    ...   
202008월                   0.7      2326.17        1187.94  1221.939746  113.1   
202009월                   1.0      2327.89        1164.65  1208.597571  114.0   
202010월                   0.1      2267.15        1136.49  1001.099665  115.6   
202011월                   0.6      2591.34        1109.32   919.546249  118.3   
202012월                   0.5      2873.47        1084.47  1110.592804  121.4   

         sieon_sent_score  
Date                       
201101월     -1.635618e+08  
201102월     -1.777621e+08  
201103월     -1.980306e+08  
201104월     -1.898849e+08  
201105월     -2.402702e+08  
...                   ...  
202008월      3.239718e+07  
202009월      4.511543e+07  
202010월      4.169356e+07  
202011월      4.231788e+07  
202012월      5.645645e+07  

[120 rows x 10 columns]

In [55]:
final_df.to_csv('df_0507.csv')

In [56]:
a = pd.read_csv('df_0507.csv')

In [57]:
a

Date    sent_score      M2  comp basemoney rate  base rate  \
0    201101월 -6.268386e+07  1676.4                 4.52       2.75   
1    201102월 -6.255832e+07  1674.4                 4.72       2.75   
2    201103월 -6.957441e+07  1677.5                 4.54       3.00   
3    201104월 -7.305733e+07  1684.8                 4.54       3.00   
4    201105월 -9.356633e+07  1690.5                 4.44       3.00   
..       ...           ...     ...                  ...        ...   
115  202008월 -5.824726e+07  3100.4                 2.19       0.50   
116  202009월 -6.441351e+07  3115.2                 2.24       0.50   
117  202010월 -3.751069e+07  3152.8                 2.24       0.50   
118  202011월 -3.147741e+07  3183.5                 2.25       0.50   
119  202012월 -3.986893e+07  3183.5                 2.21       0.50   

     consumer price index  kospi close  ex-rate close       효준감성점수    spi  \
0                     3.4      2069.73        1119.25  1856.746125   81.3   
1                     3.9      1939.30        1124.65  2046.702906   82.5   
2                     4.1      2106.70        1095.50  1715.795289   83.4   
3                     3.8      2192.36        1068.00  1635.262791   83.9   
4                     3.9      2142.47        1078.00  1922.518709   84.1   
..                    ...          ...            ...          ...    ...   
115                   0.7      2326.17        1187.94  1221.939746  113.1   
116                   1.0      2327.89        1164.65  1208.597571  114.0   
117                   0.1      2267.15        1136.49  1001.099665  115.6   
118                   0.6      2591.34        1109.32   919.546249  118.3   
119                   0.5      2873.47        1084.47  1110.592804  121.4   

     sieon_sent_score  
0       -1.635618e+08  
1       -1.777621e+08  
2       -1.980306e+08  
3       -1.898849e+08  
4       -2.402702e+08  
..                ...  
115      3.239718e+07  
116      4.511543e+07  
117      4.169356e+07  
118      4.231788e+07  
119      5.645645e+07  

[120 rows x 11 columns]

# 전처리

In [12]:
df['next_spi'] = df['spi'].shift(-1)
df['next_spi'][-1] = 124.9

In [122]:
#전처리에 사용되었던 더미 변수들 drop
df.drop(['spi'], axis = 1, inplace = True)

In [123]:
X = df.copy()
y = X.pop('next_spi')

In [124]:
X

sent_score      M2  comp basemoney rate  base rate  \
Date                                                            
201101월 -6.268386e+07  1676.4                 4.52       2.75   
201102월 -6.255832e+07  1674.4                 4.72       2.75   
201103월 -6.957441e+07  1677.5                 4.54       3.00   
201104월 -7.305733e+07  1684.8                 4.54       3.00   
201105월 -9.356633e+07  1690.5                 4.44       3.00   
...               ...     ...                  ...        ...   
202008월 -5.824726e+07  3100.4                 2.19       0.50   
202009월 -6.441351e+07  3115.2                 2.24       0.50   
202010월 -3.751069e+07  3152.8                 2.24       0.50   
202011월 -3.147741e+07  3183.5                 2.25       0.50   
202012월 -3.986893e+07  3183.5                 2.21       0.50   

         consumer price index  kospi close  ex-rate close       효준감성점수  
Date                                                                    
201101월                   3.4      2069.73        1119.25  1856.746125  
201102월                   3.9      1939.30        1124.65  2046.702906  
201103월                   4.1      2106.70        1095.50  1715.795289  
201104월                   3.8      2192.36        1068.00  1635.262791  
201105월                   3.9      2142.47        1078.00  1922.518709  
...                       ...          ...            ...          ...  
202008월                   0.7      2326.17        1187.94  1221.939746  
202009월                   1.0      2327.89        1164.65  1208.597571  
202010월                   0.1      2267.15        1136.49  1001.099665  
202011월                   0.6      2591.34        1109.32   919.546249  
202012월                   0.5      2873.47        1084.47  1110.592804  

[120 rows x 8 columns]

In [125]:
y

Date
201101월     82.5
201102월     83.4
201103월     83.9
201104월     84.1
201105월     84.4
           ...  
202008월    114.0
202009월    115.6
202010월    118.3
202011월    121.4
202012월    124.9
Name: next_spi, Length: 120, dtype: float64

In [126]:
from sklearn.model_selection import train_test_split

In [127]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.25, random_state = 0)

In [128]:
# 거시경제변수만 포함된 X 만들기
X_train_basic = X_train.drop(['sent_score','효준감성점수'], axis = 1)
X_test_basic = X_test.drop(['sent_score','효준감성점수'], axis = 1)

In [129]:
X_train_basic

M2  comp basemoney rate  base rate  consumer price index  \
Date                                                                    
201504월  2148.1                 1.99       1.75                   0.4   
202006월  3077.3                 2.22       0.50                   0.0   
201912월  2912.0                 1.95       1.25                   0.7   
201602월  2285.3                 1.95       1.50                   1.1   
201104월  1684.8                 4.54       3.00                   3.8   
...         ...                  ...        ...                   ...   
201608월  2377.3                 1.65       1.25                   0.5   
201605월  2312.8                 1.91       1.50                   0.8   
202010월  3152.8                 2.24       0.50                   0.1   
201412월  2088.7                 2.46       2.00                   0.8   
201409월  2037.6                 2.78       2.25                   1.1   

         kospi close  ex-rate close  
Date                                 
201504월      2127.17        1076.34  
202006월      2108.33        1199.28  
201912월      2197.67        1154.07  
201602월      1916.66        1237.24  
201104월      2192.36        1068.00  
...              ...            ...  
201608월      2034.65        1115.29  
201605월      1983.40        1188.39  
202010월      2267.15        1136.49  
201412월      1915.59        1092.90  
201409월      2020.09        1054.55  

[90 rows x 6 columns]

# 시언 감성 점수 Case

## 시언 감성 점수 제외한 거시경제변수만 포함한 성능
=> 효준 감성 점수 제외한 거시경제변수만 포함한 성능이랑 동일함

**kernel 별 RMSE 값** ( 소숫점 다섯째 자리부터 버림 )
- linear : 2.6694
- poly : 4.9715
- rbf : 5.8788
- sigmoid : 3.6421

## 시언 감성 점수 포함 성능

In [139]:
X_train_si = X_train.drop('효준감성점수',axis = 1)
X_test_si = X_test.drop('효준감성점수',axis = 1)

In [140]:
X_train_si

sent_score      M2  comp basemoney rate  base rate  \
Date                                                            
201504월 -7.749431e+07  2148.1                 1.99       1.75   
202006월 -5.668748e+07  3077.3                 2.22       0.50   
201912월 -6.800406e+07  2912.0                 1.95       1.25   
201602월 -6.456685e+07  2285.3                 1.95       1.50   
201104월 -7.305733e+07  1684.8                 4.54       3.00   
...               ...     ...                  ...        ...   
201608월 -5.885133e+07  2377.3                 1.65       1.25   
201605월 -5.930643e+07  2312.8                 1.91       1.50   
202010월 -3.751069e+07  3152.8                 2.24       0.50   
201412월 -8.014187e+07  2088.7                 2.46       2.00   
201409월 -9.049827e+07  2037.6                 2.78       2.25   

         consumer price index  kospi close  ex-rate close  
Date                                                       
201504월                   0.4      2127.17        1076.34  
202006월                   0.0      2108.33        1199.28  
201912월                   0.7      2197.67        1154.07  
201602월                   1.1      1916.66        1237.24  
201104월                   3.8      2192.36        1068.00  
...                       ...          ...            ...  
201608월                   0.5      2034.65        1115.29  
201605월                   0.8      1983.40        1188.39  
202010월                   0.1      2267.15        1136.49  
201412월                   0.8      1915.59        1092.90  
201409월                   1.1      2020.09        1054.55  

[90 rows x 7 columns]

In [141]:
X_test_si

sent_score      M2  comp basemoney rate  base rate  \
Date                                                            
201501월 -7.296448e+07  2092.2                 2.36       2.00   
201811월 -8.226837e+07  2693.0                 2.40       1.75   
201812월 -9.637543e+07  2700.0                 2.30       1.75   
201109월 -1.038473e+08  1729.5                 4.24       3.25   
201902월 -1.019637e+08  2730.0                 2.25       1.75   
201211월 -1.538404e+08  1830.3                 3.32       2.75   
201108월 -9.440600e+07  1719.4                 4.29       3.25   
201111월 -7.628086e+07  1753.3                 4.24       3.25   
201410월 -7.599657e+07  2051.1                 2.60       2.00   
201806월 -7.884006e+07  2622.3                 2.81       1.50   
201310월 -1.357984e+08  1908.6                 3.25       2.50   
201503월 -6.757694e+07  2127.9                 2.13       1.75   
201103월 -6.957441e+07  1677.5                 4.54       3.00   
201601월 -7.717565e+07  2261.4                 2.10       1.50   
202012월 -3.986893e+07  3183.5                 2.21       0.50   
201703월 -6.644840e+07  2437.0                 2.21       1.25   
201307월 -1.402025e+08  1890.7                 3.38       2.50   
201408월 -1.059379e+08  2031.8                 2.93       2.25   
202004월 -4.641345e+07  3015.8                 2.15       0.75   
201705월 -6.198479e+07  2454.4                 2.22       1.25   
201604월 -6.512726e+07  2299.1                 1.94       1.50   
201512월 -7.914321e+07  2246.1                 2.14       1.50   
201205월 -1.371251e+08  1784.2                 4.01       3.25   
201301월 -9.723206e+07  1841.1                 3.20       2.75   
202003월 -5.433844e+07  2984.3                 1.84       0.75   
201202월 -1.638701e+08  1763.0                 4.25       3.25   
201507월 -8.289522e+07  2200.5                 2.00       1.50   
201810월 -7.441711e+07  2674.5                 2.45       1.50   
201303월 -1.396918e+08  1862.4                 2.95       2.75   
201609월 -6.749483e+07  2383.0                 1.70       1.25   

         consumer price index  kospi close  ex-rate close  
Date                                                       
201501월                   1.0      1949.26        1101.40  
201811월                   2.0      2096.86        1119.44  
201812월                   1.3      2041.04        1113.30  
201109월                   3.8      1769.65        1180.90  
201902월                   0.5      2195.44        1124.60  
201211월                   1.6      1932.90        1081.50  
201108월                   4.7      1880.11        1064.90  
201111월                   4.2      1847.51        1140.05  
201410월                   1.1      1964.43        1072.60  
201806월                   1.5      2326.13        1113.91  
201310월                   0.9      2030.09        1060.83  
201503월                   0.5      2041.03        1107.99  
201103월                   4.1      2106.70        1095.50  
201601월                   0.6      1912.06        1207.11  
202012월                   0.5      2873.47        1084.47  
201703월                   2.3      2160.23        1116.65  
201307월                   1.6      1914.03        1123.25  
201408월                   1.4      2068.54        1013.85  
202004월                   0.1      1947.56        1212.94  
201705월                   2.0      2347.38        1118.65  
201604월                   1.0      1994.15        1143.83  
201512월                   1.1      1961.31        1174.95  
201205월                   2.5      1843.47        1179.87  
201301월                   1.6      1961.94        1087.31  
202003월                   1.0      1754.64        1218.30  
201202월                   3.0      2030.25        1118.80  
201507월                   0.7      2030.16        1165.67  
201810월                   2.0      2029.69        1141.02  
201303월                   1.5      2004.89        1111.05  
201609월                   1.3      2043.63        1100.91

In [148]:
sc = StandardScaler()
X_train_si = sc.fit_transform(X_train_si)
X_test_si = sc.transform(X_test_si)

In [167]:
# default parameter 
svr = SVR(kernel = 'rbf')

In [168]:
svr.fit(X_train_si, y_train)

SVR()

In [169]:
y_pred = svr.predict(X_test_si)

In [170]:
RMSE = mean_squared_error(y_test, y_pred)**0.5

In [171]:
RMSE

5.953910008241991

In [172]:
del svr, RMSE

**kernel 별 RMSE 값** ( 소숫점 다섯째 자리부터 버림 )
- 거시경제변수만 사용 ==> 시언이 감성 점수 포함
- linear : 2.6694 ==> 2.6446
- poly : 4.9715 ==> 4.4179
- rbf : 5.8788 ==> 5.9539
- sigmoid : 3.6421 ==> 3.5352

# 효준 감성 점수 Case

## 효준 감성 점수 제외한 거시경제변수만 포함한 성능

In [173]:
sc = StandardScaler()
X_train_basic = sc.fit_transform(X_train_basic)
X_test_basic = sc.transform(X_test_basic)

In [174]:
# default parameter 
svr = SVR(kernel = 'sigmoid')

In [175]:
svr.fit(X_train_basic, y_train)

SVR(kernel='sigmoid')

In [176]:
y_pred = svr.predict(X_test_basic)

In [178]:
RMSE = mean_squared_error(y_test, y_pred)**0.5

In [ ]:
RMSE

In [180]:
del svr, RMSE

**kernel 별 RMSE 값** ( 소숫점 다섯째 자리부터 버림 )
- linear : 2.6694
- poly : 4.9715
- rbf : 5.8788
- sigmoid : 3.6421

## 효준 감성 점수 포함 성능

In [181]:
X_train_hj = X_train.drop('sent_score',axis = 1)
X_test_hj = X_test.drop('sent_score',axis = 1)

In [182]:
X_train_hj

M2  comp basemoney rate  base rate  consumer price index  \
Date                                                                    
201504월  2148.1                 1.99       1.75                   0.4   
202006월  3077.3                 2.22       0.50                   0.0   
201912월  2912.0                 1.95       1.25                   0.7   
201602월  2285.3                 1.95       1.50                   1.1   
201104월  1684.8                 4.54       3.00                   3.8   
...         ...                  ...        ...                   ...   
201608월  2377.3                 1.65       1.25                   0.5   
201605월  2312.8                 1.91       1.50                   0.8   
202010월  3152.8                 2.24       0.50                   0.1   
201412월  2088.7                 2.46       2.00                   0.8   
201409월  2037.6                 2.78       2.25                   1.1   

         kospi close  ex-rate close       효준감성점수  
Date                                              
201504월      2127.17        1076.34  2212.598416  
202006월      2108.33        1199.28  1027.105971  
201912월      2197.67        1154.07   873.706855  
201602월      1916.66        1237.24  1422.961939  
201104월      2192.36        1068.00  1635.262791  
...              ...            ...          ...  
201608월      2034.65        1115.29  1454.840123  
201605월      1983.40        1188.39  1619.125596  
202010월      2267.15        1136.49  1001.099665  
201412월      1915.59        1092.90  1959.479960  
201409월      2020.09        1054.55  2410.587704  

[90 rows x 7 columns]

In [183]:
X_test_hj

M2  comp basemoney rate  base rate  consumer price index  \
Date                                                                    
201501월  2092.2                 2.36       2.00                   1.0   
201811월  2693.0                 2.40       1.75                   2.0   
201812월  2700.0                 2.30       1.75                   1.3   
201109월  1729.5                 4.24       3.25                   3.8   
201902월  2730.0                 2.25       1.75                   0.5   
201211월  1830.3                 3.32       2.75                   1.6   
201108월  1719.4                 4.29       3.25                   4.7   
201111월  1753.3                 4.24       3.25                   4.2   
201410월  2051.1                 2.60       2.00                   1.1   
201806월  2622.3                 2.81       1.50                   1.5   
201310월  1908.6                 3.25       2.50                   0.9   
201503월  2127.9                 2.13       1.75                   0.5   
201103월  1677.5                 4.54       3.00                   4.1   
201601월  2261.4                 2.10       1.50                   0.6   
202012월  3183.5                 2.21       0.50                   0.5   
201703월  2437.0                 2.21       1.25                   2.3   
201307월  1890.7                 3.38       2.50                   1.6   
201408월  2031.8                 2.93       2.25                   1.4   
202004월  3015.8                 2.15       0.75                   0.1   
201705월  2454.4                 2.22       1.25                   2.0   
201604월  2299.1                 1.94       1.50                   1.0   
201512월  2246.1                 2.14       1.50                   1.1   
201205월  1784.2                 4.01       3.25                   2.5   
201301월  1841.1                 3.20       2.75                   1.6   
202003월  2984.3                 1.84       0.75                   1.0   
201202월  1763.0                 4.25       3.25                   3.0   
201507월  2200.5                 2.00       1.50                   0.7   
201810월  2674.5                 2.45       1.50                   2.0   
201303월  1862.4                 2.95       2.75                   1.5   
201609월  2383.0                 1.70       1.25                   1.3   

         kospi close  ex-rate close       효준감성점수  
Date                                              
201501월      1949.26        1101.40  1850.441197  
201811월      2096.86        1119.44  1693.663795  
201812월      2041.04        1113.30  1645.223765  
201109월      1769.65        1180.90  1915.112598  
201902월      2195.44        1124.60  1629.146004  
201211월      1932.90        1081.50  2333.739232  
201108월      1880.11        1064.90  2334.489846  
201111월      1847.51        1140.05  1103.045565  
201410월      1964.43        1072.60  1950.946648  
201806월      2326.13        1113.91  1602.940430  
201310월      2030.09        1060.83  2933.526015  
201503월      2041.03        1107.99  2283.067495  
201103월      2106.70        1095.50  1715.795289  
201601월      1912.06        1207.11  1779.896207  
202012월      2873.47        1084.47  1110.592804  
201703월      2160.23        1116.65  1343.983779  
201307월      1914.03        1123.25  3607.017572  
201408월      2068.54        1013.85  3285.332083  
202004월      1947.56        1212.94   641.191318  
201705월      2347.38        1118.65  1460.088658  
201604월      1994.15        1143.83  1771.682676  
201512월      1961.31        1174.95  1976.713145  
201205월      1843.47        1179.87  2259.038046  
201301월      1961.94        1087.31  2221.292395  
202003월      1754.64        1218.30  1055.354974  
201202월      2030.25        1118.80  2857.557608  
201507월      2030.16        1165.67  2636.101339  
201810월      2029.69        1141.02  1828.420124  
201303월      2004.89        1111.05  3738.711179  
201609월      2043.63        1100.91  1450.710929

In [184]:
sc = StandardScaler()
X_train_hj = sc.fit_transform(X_train_hj)
X_test_hj = sc.transform(X_test_hj)

In [185]:
# default parameter 
svr = SVR(kernel = 'sigmoid')

In [186]:
svr.fit(X_train_hj, y_train)

SVR(kernel='sigmoid')

In [187]:
y_pred = svr.predict(X_test_hj)

In [188]:
RMSE = mean_squared_error(y_test, y_pred)**0.5

In [189]:
RMSE

3.64668654835819

In [190]:
del svr, RMSE

**kernel 별 RMSE 값** ( 소숫점 다섯째 자리부터 버림 )
- linear : 2.6694 ==> 2.5692
- poly : 4.9715 ==> 3.9581
- rbf : 5.8788 ==> 5.9429
- sigmoid : 3.6421 ==> 3.6466